pkg/engine/planner/logical/planner.go

```diff
@@ -22,6 +46,11 @@ func BuildPlan(query logql.Params) (*Plan, error) {
 	var err error
 
 	expr := query.GetExpression()
+
+	if !canExecuteWithNewEngine(expr) {
+		return nil, errors.New("query contains unimplemented features")
+	}
```

Comment:
I'm concerned that it might be annoying to keep this in sync with the `expr.Walk` below; should that track whether you've gotten an acceptable query?